### ISSUE TITLE
Characterization

### ISSUE THREAD

**Issue by pbr6cornell** [OHDSI/Atlas#1427]
### Expected behavior

Covariates should be uniquely represented in table, no redundant records

### Actual behavior

Redundant covariates observed.  Likely some issue with SQL join.  Confirm that joins are on covariate_id and not covariate_name,

Example results showing redundant records from drug domain:

![image](https://user-images.githubusercontent.com/731930/53926410-52f7a200-4051-11e9-8d1d-9514d53c1e24.png)

### Steps to reproduce behavior

Review results from characterization with >=2 cohorts and using Features using Drug Group Era.

**Comment by pavgra** [comment]
@pbr6cornell , would you mind sharing your Characterization design? (Utilities -> Export)

**Comment by pbr6cornell** [comment]
{
  "name": "[ONC] Characterization of multiple myeloma treatment initiators",
  "cohorts": [
    {
      "id": 5477,
      "name": "[ONC] New users of Daratumumab - for characterization",
      "expression": {
        "PrimaryCriteria": {
          "CriteriaList": [
            {
              "DrugExposure": {
                "CodesetId": 0,
                "First": true,
                "DrugTypeExclude": false
              }
            }
          ],
          "ObservationWindow": {
            "PriorDays": 365,
            "PostDays": 0
          },
          "PrimaryCriteriaLimit": {
            "Type": "First"
          }
        },
        "AdditionalCriteria": {
          "Type": "ALL",
          "CriteriaList": [
            {
              "Criteria": {
                "ConditionOccurrence": {
                  "CodesetId": 1,
                  "ConditionTypeExclude": false
                }
              },
              "StartWindow": {
                "Start": {
                  "Coeff": -1
                },
                "End": {
                  "Days": 0,
                  "Coeff": 1
                },
                "UseEventEnd": false
              },
              "RestrictVisit": false,
              "IgnoreObservationPeriod": false,
              "Occurrence": {
                "Type": 2,
                "Count": 1,
                "IsDistinct": false
              }
            }
          ],
          "DemographicCriteriaList": [],
          "Groups": []
        },
        "ConceptSets": [
          {
            "id": 0,
            "name": "[MLMM] Daratumumab",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 35605744,
                    "CONCEPT_NAME": "daratumumab",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "1721947",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          },
          {
            "id": 1,
            "name": "[LLH] Multiple myeloma (includes remitted/relapsed)",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 437233,
                    "CONCEPT_NAME": "Multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "109989006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 42538151,
                    "CONCEPT_NAME": "Osteoporosis co-occurrent and due to multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "739301006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 133154,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "95210003",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 36402498,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "9733/3-C42.0",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "ICDO3",
                    "CONCEPT_CLASS_ID": "ICDO Condition"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          }
        ],
        "QualifiedLimit": {
          "Type": "First"
        },
        "ExpressionLimit": {
          "Type": "First"
        },
        "InclusionRules": [],
        "EndStrategy": {
          "CustomEra": {
            "DrugCodesetId": 0,
            "GapDays": 90,
            "Offset": 0
          }
        },
        "CensoringCriteria": [],
        "CollapseSettings": {
          "CollapseType": "ERA",
          "EraPad": 90
        },
        "CensorWindow": {}
      }
    },
    {
      "id": 5478,
      "name": "[ONC] New user of Bortezomib - for characterization",
      "expression": {
        "PrimaryCriteria": {
          "CriteriaList": [
            {
              "DrugExposure": {
                "CodesetId": 0,
                "First": true,
                "OccurrenceStartDate": {
                  "Value": "2016-11-22",
                  "Op": "gt"
                },
                "DrugTypeExclude": false
              }
            }
          ],
          "ObservationWindow": {
            "PriorDays": 365,
            "PostDays": 0
          },
          "PrimaryCriteriaLimit": {
            "Type": "First"
          }
        },
        "ConceptSets": [
          {
            "id": 0,
            "name": "[MLMM] Bortezomib",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 1336825,
                    "CONCEPT_NAME": "bortezomib",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "358258",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          },
          {
            "id": 1,
            "name": "[LLH] Multiple myeloma (includes remitted/relapsed)",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 437233,
                    "CONCEPT_NAME": "Multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "109989006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 42538151,
                    "CONCEPT_NAME": "Osteoporosis co-occurrent and due to multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "739301006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 133154,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "95210003",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 36402498,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "9733/3-C42.0",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "ICDO3",
                    "CONCEPT_CLASS_ID": "ICDO Condition"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          },
          {
            "id": 2,
            "name": "[MLMM] Daratumumab",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 35605744,
                    "CONCEPT_NAME": "daratumumab",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "1721947",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          }
        ],
        "QualifiedLimit": {
          "Type": "First"
        },
        "ExpressionLimit": {
          "Type": "First"
        },
        "InclusionRules": [
          {
            "name": "has multiple myeloma",
            "expression": {
              "Type": "ALL",
              "CriteriaList": [
                {
                  "Criteria": {
                    "ConditionOccurrence": {
                      "CodesetId": 1,
                      "ConditionTypeExclude": false
                    }
                  },
                  "StartWindow": {
                    "Start": {
                      "Coeff": -1
                    },
                    "End": {
                      "Days": 0,
                      "Coeff": 1
                    },
                    "UseEventEnd": false
                  },
                  "RestrictVisit": false,
                  "IgnoreObservationPeriod": false,
                  "Occurrence": {
                    "Type": 2,
                    "Count": 1,
                    "IsDistinct": false
                  }
                }
              ],
              "DemographicCriteriaList": [],
              "Groups": []
            }
          },
          {
            "name": "does not have prior dara use (including thru next 7 days)",
            "expression": {
              "Type": "ALL",
              "CriteriaList": [
                {
                  "Criteria": {
                    "DrugExposure": {
                      "CodesetId": 2,
                      "DrugTypeExclude": false
                    }
                  },
                  "StartWindow": {
                    "Start": {
                      "Coeff": -1
                    },
                    "End": {
                      "Days": 7,
                      "Coeff": 1
                    },
                    "UseEventEnd": false
                  },
                  "RestrictVisit": false,
                  "IgnoreObservationPeriod": false,
                  "Occurrence": {
                    "Type": 0,
                    "Count": 0,
                    "IsDistinct": false
                  }
                }
              ],
              "DemographicCriteriaList": [],
              "Groups": []
            }
          }
        ],
        "EndStrategy": {
          "CustomEra": {
            "DrugCodesetId": 0,
            "GapDays": 90,
            "Offset": 0
          }
        },
        "CensoringCriteria": [],
        "CollapseSettings": {
          "CollapseType": "ERA",
          "EraPad": 90
        },
        "CensorWindow": {}
      }
    },
    {
      "id": 5479,
      "name": " [ONC] New users of Lenalidomide - for characterization",
      "expression": {
        "PrimaryCriteria": {
          "CriteriaList": [
            {
              "DrugExposure": {
                "CodesetId": 0,
                "First": true,
                "OccurrenceStartDate": {
                  "Value": "2016-11-22",
                  "Op": "gt"
                },
                "DrugTypeExclude": false
              }
            }
          ],
          "ObservationWindow": {
            "PriorDays": 365,
            "PostDays": 0
          },
          "PrimaryCriteriaLimit": {
            "Type": "First"
          }
        },
        "ConceptSets": [
          {
            "id": 0,
            "name": "[MLMM] Lenalidomide",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 19026972,
                    "CONCEPT_NAME": "lenalidomide",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "342369",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          },
          {
            "id": 1,
            "name": "[LLH] Multiple myeloma (includes remitted/relapsed)",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 437233,
                    "CONCEPT_NAME": "Multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "109989006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 42538151,
                    "CONCEPT_NAME": "Osteoporosis co-occurrent and due to multiple myeloma",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "739301006",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 133154,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "95210003",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "SNOMED",
                    "CONCEPT_CLASS_ID": "Clinical Finding"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                },
                {
                  "concept": {
                    "CONCEPT_ID": 36402498,
                    "CONCEPT_NAME": "Plasma cell leukemia",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "9733/3-C42.0",
                    "DOMAIN_ID": "Condition",
                    "VOCABULARY_ID": "ICDO3",
                    "CONCEPT_CLASS_ID": "ICDO Condition"
                  },
                  "isExcluded": true,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          },
          {
            "id": 2,
            "name": "[MLMM] Daratumumab",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 35605744,
                    "CONCEPT_NAME": "daratumumab",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "1721947",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          }
        ],
        "QualifiedLimit": {
          "Type": "First"
        },
        "ExpressionLimit": {
          "Type": "First"
        },
        "InclusionRules": [
          {
            "name": "has multiple myeloma",
            "expression": {
              "Type": "ALL",
              "CriteriaList": [
                {
                  "Criteria": {
                    "ConditionOccurrence": {
                      "CodesetId": 1,
                      "ConditionTypeExclude": false
                    }
                  },
                  "StartWindow": {
                    "Start": {
                      "Coeff": -1
                    },
                    "End": {
                      "Days": 0,
                      "Coeff": 1
                    },
                    "UseEventEnd": false
                  },
                  "RestrictVisit": false,
                  "IgnoreObservationPeriod": false,
                  "Occurrence": {
                    "Type": 2,
                    "Count": 1,
                    "IsDistinct": false
                  }
                }
              ],
              "DemographicCriteriaList": [],
              "Groups": []
            }
          },
          {
            "name": "has nor prior dara",
            "expression": {
              "Type": "ALL",
              "CriteriaList": [
                {
                  "Criteria": {
                    "DrugExposure": {
                      "CodesetId": 2,
                      "DrugTypeExclude": false
                    }
                  },
                  "StartWindow": {
                    "Start": {
                      "Coeff": -1
                    },
                    "End": {
                      "Days": 7,
                      "Coeff": 1
                    },
                    "UseEventEnd": false
                  },
                  "RestrictVisit": false,
                  "IgnoreObservationPeriod": false,
                  "Occurrence": {
                    "Type": 0,
                    "Count": 0,
                    "IsDistinct": false
                  }
                }
              ],
              "DemographicCriteriaList": [],
              "Groups": []
            }
          }
        ],
        "EndStrategy": {
          "CustomEra": {
            "DrugCodesetId": 0,
            "GapDays": 90,
            "Offset": 0
          }
        },
        "CensoringCriteria": [],
        "CollapseSettings": {
          "CollapseType": "ERA",
          "EraPad": 90
        },
        "CensorWindow": {}
      }
    }
  ],
  "featureAnalyses": [
    {
      "id": 5,
      "type": "PRESET",
      "name": "Drug Group Era Long Term",
      "domain": "DRUG",
      "descr": "One covariate per drug rolled up to ATC groups in the drug_era table overlapping with any part of the long term window.",
      "design": "DrugGroupEraLongTerm",
      "description": "One covariate per drug rolled up to ATC groups in the drug_era table overlapping with any part of the long term window.",
      "statType": "PREVALENCE"
    },
    {
      "id": 30,
      "type": "PRESET",
      "name": "Condition Group Era Long Term",
      "domain": "CONDITION",
      "descr": "One covariate per condition era rolled up to groups in the condition_era table overlapping with any part of the long term window.",
      "design": "ConditionGroupEraLongTerm",
      "description": "One covariate per condition era rolled up to groups in the condition_era table overlapping with any part of the long term window.",
      "statType": "PREVALENCE"
    },
    {
      "id": 41,
      "type": "PRESET",
      "name": "Distinct Condition Count Long Term",
      "domain": "CONDITION",
      "descr": "The number of distinct condition concepts observed in the long term window.",
      "design": "DistinctConditionCountLongTerm",
      "description": "The number of distinct condition concepts observed in the long term window.",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 46,
      "type": "PRESET",
      "name": "Distinct Ingredient Count Long Term",
      "domain": "DRUG",
      "descr": "The number of distinct ingredients observed in the long term window.",
      "design": "DistinctIngredientCountLongTerm",
      "description": "The number of distinct ingredients observed in the long term window.",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 49,
      "type": "PRESET",
      "name": "Charlson Index",
      "domain": "CONDITION",
      "descr": "The Charlson comorbidity index (Romano adaptation) using all conditions prior to the window end.",
      "design": "CharlsonIndex",
      "description": "The Charlson comorbidity index (Romano adaptation) using all conditions prior to the window end.",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 60,
      "type": "PRESET",
      "name": "Measurement Long Term",
      "domain": "MEASUREMENT",
      "descr": "One covariate per measurement in the measurement table in the long term window.",
      "design": "MeasurementLongTerm",
      "description": "One covariate per measurement in the measurement table in the long term window.",
      "statType": "PREVALENCE"
    },
    {
      "id": 66,
      "type": "PRESET",
      "name": "Distinct Procedure Count Long Term",
      "domain": "PROCEDURE",
      "descr": "The number of distinct procedures observed in the long term window.",
      "design": "DistinctProcedureCountLongTerm",
      "description": "The number of distinct procedures observed in the long term window.",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 68,
      "type": "PRESET",
      "name": "Demographics Index Month",
      "domain": "DEMOGRAPHICS",
      "descr": "Month of the index date.",
      "design": "DemographicsIndexMonth",
      "description": "Month of the index date.",
      "statType": "PREVALENCE"
    },
    {
      "id": 71,
      "type": "PRESET",
      "name": "Demographics Age Group",
      "domain": "DEMOGRAPHICS",
      "descr": "Age of the subject on the index date (in 5 year age groups)",
      "design": "DemographicsAgeGroup",
      "description": "Age of the subject on the index date (in 5 year age groups)",
      "statType": "PREVALENCE"
    },
    {
      "id": 74,
      "type": "PRESET",
      "name": "Demographics Gender",
      "domain": "DEMOGRAPHICS",
      "descr": "Gender of the subject.",
      "design": "DemographicsGender",
      "description": "Gender of the subject.",
      "statType": "PREVALENCE"
    },
    {
      "id": 77,
      "type": "PRESET",
      "name": "Demographics Age",
      "domain": "DEMOGRAPHICS",
      "descr": "Age of the subject on the index date (in years).",
      "design": "DemographicsAge",
      "description": "Age of the subject on the index date (in years).",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 80,
      "type": "PRESET",
      "name": "Demographics Index Year",
      "domain": "DEMOGRAPHICS",
      "descr": "Year of the index date.",
      "design": "DemographicsIndexYear",
      "description": "Year of the index date.",
      "statType": "PREVALENCE"
    },
    {
      "id": 84,
      "type": "PRESET",
      "name": "Procedure Occurrence Long Term",
      "domain": "PROCEDURE",
      "descr": "One covariate per procedure in the procedure_occurrence table in the long term window.",
      "design": "ProcedureOccurrenceLongTerm",
      "description": "One covariate per procedure in the procedure_occurrence table in the long term window.",
      "statType": "PREVALENCE"
    },
    {
      "id": 95,
      "type": "PRESET",
      "name": "Distinct Measurement Count Long Term",
      "domain": "MEASUREMENT",
      "descr": "The number of distinct measurements observed in the long term window.",
      "design": "DistinctMeasurementCountLongTerm",
      "description": "The number of distinct measurements observed in the long term window.",
      "statType": "DISTRIBUTION"
    },
    {
      "id": 103,
      "type": "PRESET",
      "name": "Visit Count Long Term",
      "domain": "VISIT",
      "descr": "The number of visits observed in the long term window.",
      "design": "VisitCountLongTerm",
      "description": "The number of visits observed in the long term window.",
      "statType": "DISTRIBUTION"
    }
  ],
  "parameters": [],
  "stratas": [
    {
      "name": "has prior lenalidomide",
      "criteria": {
        "Type": "ALL",
        "CriteriaList": [
          {
            "Criteria": {
              "DrugExposure": {
                "CodesetId": 0,
                "DrugTypeExclude": false
              }
            },
            "StartWindow": {
              "Start": {
                "Coeff": -1
              },
              "End": {
                "Days": 7,
                "Coeff": -1
              },
              "UseIndexEnd": false,
              "UseEventEnd": false
            },
            "RestrictVisit": false,
            "IgnoreObservationPeriod": false,
            "Occurrence": {
              "Type": 2,
              "Count": 1,
              "IsDistinct": false
            }
          }
        ],
        "DemographicCriteriaList": [],
        "Groups": []
      }
    }
  ],
  "createdAt": "2019-03-06 21:52:25",
  "stratifiedBy": "Stratified by lenalidomide use",
  "strataOnly": false,
  "strataConceptSets": [
    {
      "id": 0,
      "name": "[LLH] Test - Lenalidomide",
      "expression": {
        "items": [
          {
            "concept": {
              "CONCEPT_ID": 19026972,
              "CONCEPT_NAME": "lenalidomide",
              "STANDARD_CONCEPT": "S",
              "STANDARD_CONCEPT_CAPTION": "Standard",
              "INVALID_REASON": "V",
              "INVALID_REASON_CAPTION": "Valid",
              "CONCEPT_CODE": "342369",
              "DOMAIN_ID": "Drug",
              "VOCABULARY_ID": "RxNorm",
              "CONCEPT_CLASS_ID": "Ingredient"
            },
            "isExcluded": false,
            "includeDescendants": true,
            "includeMapped": false
          }
        ]
      }
    }
  ]
}

**Comment by pavgra** [comment]
Reproduced the issue on SynPUF:
![image](https://user-images.githubusercontent.com/10010071/53928683-01ebac00-4059-11e9-918f-2333e0ae6563.png)

Design:
```
{
  "name": "[ONC] Characterization of multiple myeloma treatment initiators",
  "cohorts": [
    {
      "id": 3,
      "name": "Ibuprofen",
      "expression": {
        "PrimaryCriteria": {
          "CriteriaList": [
            {
              "DrugExposure": {
                "CodesetId": 0,
                "DrugTypeExclude": false
              }
            }
          ],
          "ObservationWindow": {
            "PriorDays": 0,
            "PostDays": 0
          },
          "PrimaryCriteriaLimit": {
            "Type": "First"
          }
        },
        "ConceptSets": [
          {
            "id": 0,
            "name": "Ibuprofen",
            "expression": {
              "items": [
                {
                  "concept": {
                    "CONCEPT_ID": 1177480,
                    "CONCEPT_NAME": "Ibuprofen",
                    "STANDARD_CONCEPT": "S",
                    "STANDARD_CONCEPT_CAPTION": "Standard",
                    "INVALID_REASON": "V",
                    "INVALID_REASON_CAPTION": "Valid",
                    "CONCEPT_CODE": "5640",
                    "DOMAIN_ID": "Drug",
                    "VOCABULARY_ID": "RxNorm",
                    "CONCEPT_CLASS_ID": "Ingredient"
                  },
                  "isExcluded": false,
                  "includeDescendants": true,
                  "includeMapped": false
                }
              ]
            }
          }
        ],
        "QualifiedLimit": {
          "Type": "First"
        },
        "ExpressionLimit": {
          "Type": "First"
        },
        "InclusionRules": [],
        "CensoringCriteria": [],
        "CollapseSettings": {
          "CollapseType": "ERA",
          "EraPad": 0
        },
        "CensorWindow": {}
      }
    }
  ],
  "featureAnalyses": [
    {
      "id": 5,
      "type": "PRESET",
      "name": "Drug Group Era Long Term",
      "domain": "DRUG",
      "descr": "One covariate per drug rolled up to ATC groups in the drug_era table overlapping with any part of the long term window.",
      "design": "DrugGroupEraLongTerm",
      "description": "One covariate per drug rolled up to ATC groups in the drug_era table overlapping with any part of the long term window.",
      "statType": "PREVALENCE"
    }
  ],
  "parameters": [],
  "stratas": [
    {
      "name": "Test",
      "criteria": {
        "Type": "ALL",
        "CriteriaList": [
          {
            "Criteria": {
              "DrugEra": {}
            },
            "StartWindow": {
              "Start": {
                "Coeff": -1
              },
              "End": {
                "Days": 7,
                "Coeff": -1
              },
              "UseIndexEnd": false,
              "UseEventEnd": false
            },
            "RestrictVisit": false,
            "IgnoreObservationPeriod": false,
            "Occurrence": {
              "Type": 2,
              "Count": 1,
              "IsDistinct": false
            }
          }
        ],
        "DemographicCriteriaList": [],
        "Groups": []
      }
    }
  ],
  "createdAt": "2019-03-07 02:24:36",
  "strataOnly": false,
  "strataConceptSets": []
}
```

**Comment by wivern** [comment]
I guess this is not duplicated records but covariates related to different concepts having the similar names.
Probably adding concept code to covariate title would reduce some misunderstanding.
![ace_inhibitors](https://user-images.githubusercontent.com/1109753/53954528-87b63880-40e7-11e9-8132-6f889f068343.png)
![diphehydramine](https://user-images.githubusercontent.com/1109753/53954535-8d138300-40e7-11e9-948f-7a5c954b0894.png)

**Comment by anthonysena** [comment]
From my review of this item, it appears that the `Drug Group Era Long Term` feature performs some rollup to ATC which can lead to the duplicate covariate names but not duplicate data. I'd suggest that we re-format the covariate display name to: 

```
<concept_name>
(<concept_id>)
```

I think having the `concept_name` + `concept_id` will help to clear up any that may share the same name as the `concept_id` will ensure uniqueness. @pbr6cornell - what do you think of this suggestion?

**Comment by chrisknoll** [comment]
i'd prefer we separate those values into 2 distinct values.   If it's confusing on the front end, we can always show the concept ID by default.

**Comment by anthonysena** [comment]
Agreed @chrisknoll - will make it easier to sort/filter/export.

**Comment by pbr6cornell** [comment]
I'm fine to add covariate_id  or concept_id as a field to make it clear
they are not the same.     We need to make sure no joins are occurring on
the covariate name though, I think that was a bug in a prior version...

**Comment by pavgra** [comment]
Since in most cases name of covariate will match the concept name, I've added such column:
![image](https://user-images.githubusercontent.com/10010071/53995347-c60d2100-4102-11e9-9f5b-dcedc03d64ad.png)

**Comment by anthonysena** [comment]
Closed via #1436
